pkg/sdk/alerts.go

```diff
@@ -0,0 +1,382 @@
+	if opts == nil {
+		opts = &CreateAlertOptions{}
+	}
+	opts.name = id
+	opts.warehouse = warehouse
+	opts.schedule = schedule
+	opts.name = id
+	opts.condition = AlertCondition{condition}
+	opts.action = action
+	if err := opts.validate(); err != nil {
+		return err
+	}
+	sql, err := structToSQL(opts)
+	if err != nil {
+		return err
+	}
+	_, err = v.client.exec(ctx, sql)
+	return err
+}
+
+type AlertOperation string
+
+var (
+	// Resume makes a suspended alert active.
+	Resume AlertOperation = "RESUME"
+	// Suspend puts the alert into a “Suspended” state.
+	Suspend AlertOperation = "SUSPEND"
+)
+
+type AlertState string
+
+var (
+	Started   AlertState = "started"
+	Suspended AlertState = "suspended"
+)
+
+type AlterAlertOptions struct {
+	alter    bool                   `ddl:"static" sql:"ALTER"` //lint:ignore U1000 This is used in the ddl tag
+	alert    bool                   `ddl:"static" sql:"ALERT"` //lint:ignore U1000 This is used in the ddl tag
+	IfExists *bool                  `ddl:"keyword" sql:"IF EXISTS"`
+	name     SchemaObjectIdentifier `ddl:"identifier"`
+
+	// One of
+	Operation       *AlertOperation `ddl:"keyword"`
+	Set             *AlertSet       `ddl:"keyword" sql:"SET"`
+	Unset           *AlertUnset     `ddl:"keyword" sql:"UNSET"`
+	ModifyCondition *string         `ddl:"condition,parentheses" sql:"MODIFY CONDITION EXISTS"`
+	ModifyAction    *string         `ddl:"parameter,no_equals" sql:"MODIFY ACTION"`
+}
+
+func (opts *AlterAlertOptions) validate() error {
+	if !validObjectidentifier(opts.name) {
+		return errors.New("invalid object identifier")
+	}
+
+	if everyValueNil(opts.Operation, opts.Set, opts.Unset, opts.ModifyCondition, opts.ModifyAction) {
+		return errors.New("No alter action specified")
+	}
+	if !exactlyOneValueSet(opts.Operation, opts.Set, opts.Unset, opts.ModifyCondition, opts.ModifyAction) {
+		return errors.New(`
+		Only one of the following actions can be performed at a time:
+		{
+			RESUME | SUSPEND,
+			SET,
+			UNSET,
+			MODIFY CONDITION EXISTS,
+			MODIFY ACTION
+		}
+		`)
+	}
+
+	return nil
+}
+
+type AlertSet struct {
+	Warehouse *AccountObjectIdentifier `ddl:"identifier,equals" sql:"WAREHOUSE"`
+	Schedule  *string                  `ddl:"parameter,single_quotes" sql:"SCHEDULE"`
+	Comment   *string                  `ddl:"parameter,single_quotes" sql:"COMMENT"`
+}
+
+type AlertUnset struct {
+	warehouse *bool `ddl:"keyword" sql:"WAREHOUSE"`
+	schedule  *bool `ddl:"keyword" sql:"SCHEDULE"`
+	comment   *bool `ddl:"keyword" sql:"COMMENT"`
+}
+
+func (v *alerts) Alter(ctx context.Context, id SchemaObjectIdentifier, opts *AlterAlertOptions) error {
+	if opts == nil {
+		return errors.New("alter alert options cannot be empty")
+	}
+
+	opts.name = id
+	if err := opts.validate(); err != nil {
+		return err
+	}
+	sql, err := structToSQL(opts)
+	if err != nil {
+		return err
+	}
+	_, err = v.client.exec(ctx, sql)
+	return err
+}
+
+type DropAlertOptions struct {
```

Comment:
this could be make private since it shouldn't be used by outside consumers. e.g. `dropAlertOptions`